libvips/conversion/bandjoin.c

```diff
@@ -116,15 +116,38 @@ vips_bandjoin_buffer( VipsBandarySequence *seq,
 		q1 = q;
 		p1 = p[i];
 
-		for( x = 0; x < width; x++ ) {
-			for( z = 0; z < ips; z++ )
-				q1[z] = p1[z];
+		if( ips == 1 ) {
+			for( x = 0; x < width; x++ ) {
+				q1[0] = p1[x];
 
-			p1 += ips;
-			q1 += ops;
+				q1 += ops;
+			}
+
+			q += 1;
```

Comment:
```suggestion
			q += ips;
```
To make it consistent with the `else` statement, although I'm not sure if that would cause a different codegen.